agent/proxycfg/snapshot.go

```diff
@@ -831,6 +831,10 @@ func (c *configSnapshotAPIGateway) ToIngress(datacenter string) (configSnapshotI
 }
 
 func (c *configSnapshotAPIGateway) synthesizeChains(datacenter string, protocol structs.APIGatewayListenerProtocol, port int, name string, boundListener structs.BoundAPIGatewayListener) ([]structs.IngressService, structs.Upstreams, []*structs.CompiledDiscoveryChain, error) {
+	if len(boundListener.Routes) == 0 {
```

Comment:
Just moved this to the top of the function for clarity in circuit breaking.

Reply:
So, we still need to check the length of `chains` because of the conditional addition to the array. Imagine you have route refs, but you haven't triggered any of the route watches yet,  you'll wind up with an empty chain that you pass into the `Synthesize` invocation at the bottom. While it's not that big of a deal, it basically means you may be spamming logs for a bit with errors about being unable to compile empty discovery chains.

I think it may be easier just to keep the `chains` check at the bottom here.

Reply:
That's a great point! Thanks for catching that.